src/UI/Component/Panel/Listing/Divider.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/* Copyright (c) 2017 Alex Killing <killing@leifos.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Panel\Listing;
+
+/**
+ * Listing divider
+ * @package ILIAS\UI\Component\Panel\Listing
+ */
+interface Divider extends \ILIAS\UI\Component\Component {
```

Comment:
Visually it is the same thing. In Dropdowns we will also need Dividers "withLabel" which works like a Heading in HTML in both cases. And we could also extend Divider to something like Heading with an "withAction" method. For enabling actions on this level no grouping is needed. Headings in HTML also have no "knowledge" about what comes after them. There is no need for grouping on the UI level imo. An entity/data model may have a grouping interpretation of things, but I would not define it on the UI level.

Otherwise you may end up with something like this:

list->withGroupedItems("my group", $items1)
->withItems($items2)
->withAdditionalItems($items3)

or ->list("title", array($items, $item_group, $items, ...))

Which would have weak semantic and be complex in handling, imo.

Having a separate Divider/Heading component makes it much easier  to extend these, e.g. if you want to have actions on it, Heading->withActions would be quite ease to implement and understand. And actions on the Heading will act on the concept being represented by the Heading, not the single subitems (the have their own action).

And a Heading with actions can be reused in all other kind of scenarios, where you do not have grouped items. E.g. a table heading or a heading on some text.